build-scripts/clang-tidy-run.sh

```diff
@@ -7,24 +7,36 @@ set -exo pipefail
 
 num_jobs=3
 
+# enable all the switches by default
+BACKTRACE=${BACKTRACE:-1}
+LOCALIZE=${LOCALIZE:-1}
+TILES=${TILES:-1}
+SOUND=${SOUND:-1}
+
 # create compilation database (compile_commands.json)
-mkdir -p build
-cd build
-cmake \
-    -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
-    ${COMPILER:+-DCMAKE_CXX_COMPILER=$COMPILER} \
-    -DCMAKE_BUILD_TYPE="Release" \
-    -DBACKTRACE=ON \
-    -DTILES=${TILES:-0} \
-    -DSOUND=${SOUND:-0} \
-    -DLOCALIZE=${LOCALIZE:-0} \
-    ..
-cd ..
-ln -s build/compile_commands.json .
+if [ ! -f compile_commands.json ]
```

Comment:
We should still maybe run cmake always to update compile_commands.json? Cause I definitely am going to run it, forget, leave it, rebase, add new files, and wonder why they aren't getting tidy'd.

Reply:
uhh, sure, I can `ln --force` instead